eng/Directory.Build.Common.targets

```diff
@@ -264,5 +264,23 @@
   <Target Name="RunApiCompat" DependsOnTargets="ValidateApiCompatForSrc" Condition="'$(ApiCompatVersion)' != ''">
   </Target>
 
+  <Target Name="GetPackageInfo">
+    <PropertyGroup>
+      <PackageIsNewSdk>false</PackageIsNewSdk>
+      <PackageIsNewSdk Condition="'$(IsClientLibrary)' == 'true'">true</PackageIsNewSdk>
+
+      <PackageSdkType>data</PackageSdkType>
+      <PackageSdkType Condition="'$(IsClientLibrary)' == 'true'">client</PackageSdkType>
+      <PackageSdkType Condition="'$(IsMgmtLibrary)' == 'true'">mgmt</PackageSdkType>
+
+      <PackageRootDirectory>$([MSBuild]::NormalizeDirectory($(MSBuildProjectDirectory)/../).Trim("/").Trim("\\"))</PackageRootDirectory>
+
+      <!-- Parse out the service directory based on the relative path to the sdk folder -->
+      <ServiceDirectory><![CDATA[$([System.Text.RegularExpressions.Regex]::Replace($(PackageRootDirectory), '^.*[\\/]+sdk[\\/]+([^\\/]+).*$', '$1'))]]></ServiceDirectory>
+    </PropertyGroup>
+
+    <Message Condition="'$(IsShippingLibrary)' == 'true'" Text="'$(PackageRootDirectory)' '$(ServiceDirectory)' '$(PackageId)' '$(_VersionInProject)' '$(PackageSdkType)' '$(PackageIsNewSdk)'" Importance="High" />
```

Comment:
Should we do a `Key=Value` format with a line per value?

Reply:
I see, you want to run it in batch.

Reply:
Do you need quotes? I don't think they help you. If any value would have a space in it the Split(" ") would fail.

Reply:
I put the quotes in there so that if one of the properties ends up being empty then I get empty value when I split.